### PR TITLE
chore(medic/cht-monitoring#31) Extending CHT Watchdog

### PR DESCRIPTION
**Pull request by mrjones-plip**
This PR:
* Adds a "Integrating CHT Watchdog" page
* describes in detail how to add a new scrape source to watchdog
* uses cAdvisor as an example 

closes https://github.com/medic/cht-monitoring/issues/31

### CODE REVIEW

**Comment by mrjones-plip** [comment]
@jkuester  - I want to run through this guide on a clean install one more time to make sure it all works as advertised, but the resulting changes will be just a path change here or a file name change there.

**Comment by mrjones-plip** [comment]
Ok - this is finalized!  I just went through it from scratch and made a few fixes that were broken/wrong/missing.

One simplification we could make is to merge the `/home/ubuntu/cht/compose/cadvisor_compose.yml` and the `/home/ubuntu/cht/compose/caddy-compose.yml` files, but I liked that they were separate files for separate services.  Open to feedback!

**Comment by mrjones-plip** [comment]
oy!  I saw reddis was exposed on port `6379` and there's no need.  Docker containers (`reddis` <-> `cadvisor` and `caddy` <-> `cadvisor`) can see each other fine on the docker network. Only ports we need to expose here is on `caddy`. 

less ports FTW  per e6d59275

File: content/en/apps/guides/hosting/monitoring/integration.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+2. Click the upper left hamburger menu and click "Dashboards"
+3. Find the "New" button on the left, click and choose "Import" from the drop down
+4. On the next page, scroll down to find "Import via grafana.com", enter ID `193` and click "Load"
+5. Confirm the "Name" and "Folder" values and select "Prometheus" as the data source in the dropdown. Finally, click the "Import" button at the bottom of the page.
+
+That's it! After following these steps, you should be looking at the cAdvisor dashboard [as shown above](#steps-to-new-integrations).  From here, you can both customize this dashboard as well as add alerts as needed.
+
+<!-- there wasn't enough space after the diagrams, adding CSS to increase padding -->
+<style>
+.mermaid { padding-bottom: 20px; }
```
Comment (jkuester) [inline content/en/apps/guides/hosting/monitoring/integration.md:203]: Was this something specific to this page, or should we try to add this globally?

**Review by jkuester — APPROVED** [review]
Added some minor working suggestions, but all-in-all this is great!  I am stoked that it is so easy to spin up additional integrations!

File: content/en/apps/guides/hosting/monitoring/integration.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+### Steps to new integrations
+
+While this is a specific example for cAdvisor, these same steps will be taken to extend Watchdog for other metrics:
+
+1. CHT Watchdog: Adding new scrape and compose configs
+2. CHT Watchdog: Restart the Prometheus and Grafana server to include the new scrape config mounts
+3. CHT Core: Create both cAdvisor and Caddy Docker Compose files 
+4. CHT Core: Start the Caddy and a cAdvisor containers along with the CHT Core
+5. CHT Watchdog: Importing an exising cAdvisor dashboard from `grafana.com`
```
[the remark this answers is not in the document]
Comment (mrjones-plip) [reply content/en/apps/guides/hosting/monitoring/integration.md:73]: \o/ thankss!

File: content/en/apps/guides/hosting/monitoring/integration.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+Note that the CHT Upgrade Service will process all Docker Compose file in the `/home/ubuntu/cht/compose` directory for us and we don't need to explicity specify them in the `docker compose up` command.
+
+### On CHT Watchdog: Import Grafana Dashboard
+
+Now that cAdvisor is running on your CHT Core instance and CHT Watchdog's Prometheus has additional scrape configs to ingest the cAdvisor metrics, we can now visualize it in a Grafana Dashboard and then alert on it. 
+
+1. Log into your Watchdog instance
+2. Click the upper left hamburger menu and click "Dashboards"
+3. Find the "New" button on the left, click and choose "Import" from the drop down
+4. On the next page, scroll down to find "Import via grafana.com", enter ID `193` and click "Load"
```
[the remark this answers is not in the document]
Comment (mrjones-plip) [reply content/en/apps/guides/hosting/monitoring/integration.md:198]: ah - a deep link is a great idea - thank you!

File: content/en/apps/guides/hosting/monitoring/integration.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+### On CHT Watchdog
+
+#### Scrape config
+
+We'll first create the `~/cadvisor-prometheus-conf.yml` file and point the config to our CHT Core URL:
+
+```yaml
+scrape_configs:
+  - job_name: 'cadvisor'
+    scrape_interval: 1m
```
[the remark this answers is not in the document]
Comment (mrjones-plip) [reply content/en/apps/guides/hosting/monitoring/integration.md]: ah - good catch! I fixed it in 9f7c1bd.  I think I might had it higher when I was having such high CPU you load.

File: content/en/apps/guides/hosting/monitoring/integration.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+4. CHT Core: Start the Caddy and a cAdvisor containers along with the CHT Core
+5. CHT Watchdog: Importing an exising cAdvisor dashboard from `grafana.com`
+
+After completing these steps, we now have Docker metrics we can alert on:
+
+[![Screenshot of Grafana Dashboard showing data from Prometheus](cadvisor.screenshot.png)](cadvisor.screenshot.png)
+
+Read on below on how to set this up!
+
+## Additional Configuration files
```
[the remark this answers is not in the document]
Comment (mrjones-plip) [reply content/en/apps/guides/hosting/monitoring/integration.md]: agreed! fixed in 33de354

File: content/en/apps/guides/hosting/monitoring/integration.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+
+### On CHT Watchdog
+
+#### Scrape config
+
+We'll first create the `~/cadvisor-prometheus-conf.yml` file and point the config to our CHT Core URL:
```
[the remark this answers is not in the document]
Comment (mrjones-plip) [reply content/en/apps/guides/hosting/monitoring/integration.md]: fixed in 33de354

File: content/en/apps/guides/hosting/monitoring/integration.md (hunk 1 above)
Comment (mrjones-plip) [reply content/en/apps/guides/hosting/monitoring/integration.md:205]: I'll push this live now and see if it makes sense to move this globally in another PR - thanks!

Comment (mrjones-plip) [reply content/en/apps/guides/hosting/monitoring/integration.md:205]: global is a good idea!  fixing in #1083

**Pull request merged**